File: apps/web/src/lib/config/agents.ts
/**
 * Agent UI metadata
 * Shared configuration for agent display in the UI
 *
 * NOTE: This is a CLIENT-SIDE config file.
 * Keep in sync with server configs at lib/server/agents/configs.ts
 */

/**
 * UI-safe agent metadata
 * 2-agent system:
 * - agent: Has all tools, handles queries requiring data/actions (general assistant)
 * Note: onboarding agent is not shown in UI - it's automatically selected based on axiology_complete flag
 */
export const AGENT_UI_METADATA = [
	{
		id: 'agent',
		name: 'Agent',
		description: 'Intelligent assistant with access to all available tools. Can query data, search the web, visualize information, and help with tasks.',
		color: '#6b7280',
		icon: 'ri:robot-line',
		enabled: true,
	},
] as const;

/**
 * Type for agent UI metadata
 */
export type AgentUIMetadata = (typeof AGENT_UI_METADATA)[number];

/**
 * Get agent metadata by ID
 * @param id - Agent ID
 * @returns Agent metadata or undefined if not found
 */
export function getAgentUIMetadata(id: string): AgentUIMetadata | undefined {
	return AGENT_UI_METADATA.find((agent) => agent.id === id);
}

/**
 * Get all enabled agents for UI display
 * @returns Array of enabled agent metadata
 */
export function getEnabledAgents(): AgentUIMetadata[] {
	return AGENT_UI_METADATA.filter((agent) => agent.enabled);
}

File: packages/virtues-registry/src/agents.rs
//! Agent registry - Assistant personas
//!
//! Agents are static configuration defining different "modes" of the assistant.
//! Currently there is one default agent, but this could be extended.

use serde::{Deserialize, Serialize};

/// Agent configuration
#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct AgentConfig {
    /// Unique agent identifier
    pub agent_id: String,
    /// Human-readable name
    pub name: String,
    /// Description of what this agent does
    pub description: String,
    /// UI color (hex)
    pub color: String,
    /// Iconify icon name
    pub icon: String,
    /// Default model for this agent
    pub default_model: String,
    /// Maximum agentic steps before forcing completion
    pub max_steps: i32,
    /// Whether this agent is enabled
    pub enabled: bool,
    /// Sort order for UI display
    pub sort_order: i32,
}

/// Get default agent configurations
pub fn default_agents() -> Vec<AgentConfig> {
    vec![AgentConfig {
        agent_id: "agent".to_string(),
        name: "Agent".to_string(),
        description: "Intelligent assistant with access to all available tools. Can query data, search the web, visualize information, and help with tasks.".to_string(),
        color: "#6b7280".to_string(),
        icon: "ri:robot-line".to_string(),
        default_model: "anthropic/claude-sonnet-4-20250514".to_string(),
        max_steps: 5,
        enabled: true,
        sort_order: 1,
    }]
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_default_agents() {
        let agents = default_agents();
        assert!(!agents.is_empty(), "Agents should not be empty");

        // Verify all agents have required fields
        for agent in &agents {
            assert!(!agent.agent_id.is_empty());
            assert!(!agent.name.is_empty());
            assert!(!agent.default_model.is_empty());
        }
    }
}
